leaf-ui/js/displayAnalysis.js

```diff
@@ -346,19 +351,48 @@ function addAnalysisConfig(config) {
     mainElement = document.getElementById(config.id);
     mainElement.querySelector('.config-elements').addEventListener('dblclick', function(){rename(this /** Config element */)});
     mainElement.querySelector('.config-elements').addEventListener('click', function(){switchConfigs(this.parentElement /** Config element */)});
-    mainElement.querySelector('.dropdown-button').addEventListener('click', function(){toggleDropdown(this.parentElement.parentElement /** Config element */)})
+    mainElement.querySelector('.dropdown-button').addEventListener('click', function(){toggleDropdown(this.parentElement.parentElement /** Config element */)});
+    mainElement.querySelector('.deleteconfig-button').addEventListener('click', function(){ 
+        try{removeConfiguration(this.parentElement.parentElement)/** Config element */} 
+        catch(e){console.error(e); alert(e);}
+    });
+
+    // Whenever first or loaded configs are added, update the highest position 
+    // Check if configuration added has higher initialPosition than current highest position, prevents naming duplicates
+    if(config.initialPosition > highestPosition) {
+        highestPosition = config.initialPosition;
+        console.log(highestPosition);
+    }
 }
 
 /**
  * Removes an analysis configuration from the analysisMap and UI sidebar
  * TODO implement with a button
+ * @param {HTML Element} configElement
  */
-function removeConfiguration() {
+function removeConfiguration(configElement) {
+    console.log(currAnalysisConfig);
+    // Access previous and next element in HTML div, check if they exist
+    // If they do exist, set currAnalysisConfig to them
+    // If not, throw exception
+    prevElement = $(configElement).prev();
+    nextElement = $(configElement).next();
+    if ($(prevElement).length){ currAnalysisConfig = analysisMap.get(prevElement.attr("id"));
+    }else if ($(nextElement).length){ currAnalysisConfig = analysisMap.get(nextElement.attr("id"));
+    }else{
+        // Prevents undefined currAnalysisConfig
+        throw "Must have at least one configuration";
```

Comment:
Potentially allow user to delete all configs, but then repopulate with default empty Configuration1? ask @amgrubb for input